puppet/lib/puppet/util/puppetdb/char_encoding.rb

```diff
@@ -71,6 +71,20 @@ def self.utf8_string(str)
   def self.warn_if_changed(str, converted_str)
     if converted_str != str
       Puppet.warning "Ignoring invalid UTF-8 byte sequences in data to be sent to PuppetDB"
+
+      begin
+        fn = '/tmp/puppetdb-invalid-utf8-sequences'
```

Comment:
This location seems too open, if we are going to log anything it should be in a log directory of some kind. Not only that, I would think a single file for logging might not be the right direction, users might have multiple problems at different times for example, and having a single file makes that a contention for debugging properly.